tests/sweep_framework/README.md

```diff
@@ -0,0 +1,460 @@
+# Sweep Framework
+
+## Op Test Library
+
+Sweep test module files are placed in the `tests/sweep_framework/sweeps/` folder. The name of the `.py` file will be the name of the module. They use the following template:
+
+### Parameters
+
+**LIMITATION: The suites must be made up of a maximum of 10,000 individual permutations. If you wish to test on more than 10,000 vectors, split the inputs into more than one suite.**
+
+#### Example
+```
+parameters = {
+    "suite_1": {
+        "batch_sizes": [(1,)],
+        "height": [384, 1024],
+        "width": [1024, 4096],
+        "broadcast": [None, "h", "w", "hw"],
+        "input_a_dtype": [ttnn.bfloat16],
+        "input_b_dtype": [ttnn.bfloat16],
+        "input_a_layout": [ttnn.TILE_LAYOUT],
+        "input_b_layout": [ttnn.TILE_LAYOUT],
+        "input_b_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
+        "input_a_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
+        "output_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
+    },
+    "suite_2": {
+        "batch_sizes": [(1,)],
+        "height": [1024, 1024],
+        "width": [1024, 2048],
+        "broadcast": [None, "h", "hw"],
+        "input_a_dtype": [ttnn.bfloat16],
+        "input_b_dtype": [ttnn.bfloat16],
+        "input_a_layout": [ttnn.ROW_MAJOR_LAYOUT],
+        "input_b_layout": [ttnn.ROW_MAJOR_LAYOUT],
+        "input_b_memory_config": [ttnn.L1_MEMORY_CONFIG],
+        "input_a_memory_config": [ttnn.L1_MEMORY_CONFIG],
+        "output_memory_config": [
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    ttnn.CoreRangeSet(
+                        {
+                            ttnn.CoreRange(ttnn.CoreCoord(1, 3), ttnn.CoreCoord(1, 4)),
+                            ttnn.CoreRange(ttnn.CoreCoord(2, 3), ttnn.CoreCoord(2, 4)),
+                        }
+                    ),
+                    [64, 64],
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
+                ),
+            )
+        ],
+    },
+}
+```
+
+Parameters must be seperated into "suites" (in this example, suite_1 and suite_2 are the named suites). These suites allow for granularity when running test suites with the runner. A common use case may be to define post-commit, nightly, and weekly type suites where each suite is larger than the previous, and can be run less frequently.
```

Comment:
Please use vim `:set spell` or similar to go through the file. There are a bunch of typos.

Reply:
Fixed, ty